include/ia2.h

```diff
@@ -384,6 +388,19 @@ static int protect_pages(struct dl_phdr_info *info, size_t size, void *data) {
   // `SegmentInfo` will have its size field set to zero
   size_t extra_seg_info = info->dlpi_phnum;
 
+  // Add the ignored range for the DYNAMIC section. If the library is built with
+  // relro this section should be part of the relro region, so it at least won't
+  // be trivially writable without an mprotect call.
+  for (size_t i = 0; i < info->dlpi_phnum; i++) {
+    Elf64_Phdr phdr = info->dlpi_phdr[i];
+    if (phdr.p_type == PT_DYNAMIC) {
+      struct AddressRange *dynamic_range = &ignored_ranges[NUM_IGNORED_RANGES - 1];
+      dynamic_range->start = (info->dlpi_addr + phdr.p_vaddr) & ~0xFFFUL;
+      dynamic_range->end =
+          (dynamic_range->start + phdr.p_memsz + 0xFFFUL) & ~0xFFFUL;
```

Comment:
It would be clearer to at least use a `PAGE_SIZE` or `PAGE_SIZE_MASK` macro here.

Reply:
Then again, the other code here doesn't, so don't consider this a blocker.